src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.h

```diff
@@ -0,0 +1,290 @@
+appropriate
+     *   error.
+     */
+    virtual Protocols::InteractionModel::Status
+    ManuallyTriggerTransport(const uint16_t & connectionID, TriggerActivationReasonEnum activationReason,
+                             const TransportMotionTriggerTimeControlStruct & timeControl) = 0;
+
+    /**
+     *   @brief Handle Command Delegate to get the Stream Options Configuration for the specified push transport.
+     *
+     *   @param connectionID  [in]     Indicates the allocated connectionID to get the Stream Options Configuration of.
+     *
+     *   @param outtransportConfigurations  [out]     Single item list of mapped transport configuration or list if connectionID is
+     * NULL.
+     *
+     *   @return Success if the transport is already allocated; otherwise, the command SHALL be rejected with an appropriate
+     *   error.
+     *
+     */
+    virtual Protocols::InteractionModel::Status
+    FindTransport(const Optional<DataModel::Nullable<uint16_t>> & connectionID,
+                  DataModel::List<TransportConfigurationStruct> & outtransportConfigurations) = 0;
+
+    /**
+     *   @brief Delegate callback for notifying change in an attribute.
+     *
+     */
+    virtual void OnAttributeChanged(AttributeId attributeId) = 0;
+
+    /**
+     *  Delegate functions to load the allocated transport connections.
+     *  The delegate application is responsible for creating and persisting these connections ( based on the Allocation commands ).
+     *  These Load APIs would be used to load the pre-allocated transport connections context information into the cluster server
+     * list, at initialization. Once loaded, the cluster server would be serving Reads on these attributes. The list is updatable
+     * via the Add/Remove functions for the respective transport connections.
+     */
+    virtual Protocols::InteractionModel::Status LoadCurrentConnections(std::vector<uint16_t> & currentConnections) = 0;
+
+    /**
+     *  @brief Callback into the delegate once persistent attributes managed by
+     *  the Cluster have been loaded from Storage.
+     */
+    virtual Protocols::InteractionModel::Status PersistentAttributesLoadedCallback() = 0;
+
+private:
+    friend class PushAvStreamTransportServer;
+
+    PushAvStreamTransportServer * mPushAvStreamTransportServer = nullptr;
+
+    /**
+     * This method is used by the SDK to set the PushAvStreamTransportServer pointer member in the delegate.
+     * This is done in the constructor during the instantiation of the PushAvStreamTransportServer object.
+     *
+     * @param aPushAvStreamTransportServer A pointer to the PushAvStreamTransportServer object related to this delegate object.
+     */
+    void SetPushAvStreamTransportServer(PushAvStreamTransportServer * aPushAvStreamTransportServer)
+    {
+        mPushAvStreamTransportServer = aPushAvStreamTransportServer;
+    }
```

Comment:
In other cluster implementations, the cluster server is instantiated directly from the application rather than through its delegate. This approach avoids circular dependencies, since the cluster server itself contains the delegate.